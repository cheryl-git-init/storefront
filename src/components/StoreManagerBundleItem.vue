<template>
  <span>
    <span>{{itemDetails.name}}</span>
    <!-- <span v-if="itemIsInBasket">Already in your basket!</span> -->
  </span>
</template>

<script>
// @ is an alias to /src

export default {
  name: "StoreManagerBundleItem",
  components: {},
  data: function () {
    return {};
  },
  methods: {},
  mounted: function () {},
  props: {
    item: String
  },
  computed: {
    itemDetails: function () {
      return this.$store.getters.getItemInDepth(this.item);
    },
    itemIsInBasket: function () {
      return this.$store.getters.itemIsInBasket(this.item);
    }
  }
};
</script>

<style lang="scss" scoped>
</style>