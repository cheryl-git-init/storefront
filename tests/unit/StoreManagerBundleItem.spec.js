import Vuex from 'vuex';
import StoreManagerBundleItem from '@/components/StoreManagerBundleItem.vue';
import { shallowMount, createLocalVue } from '@vue/test-utils';

const localVue = createLocalVue();
localVue.use(Vuex);

describe('StoreManagerBundleItem.vue', () => {    
    const wrapper = shallowMount(StoreManagerBundleItem, {
        propsData: { 
            item: "test"
        },
        computed: { 
            itemDetails() {
                return {
                    name: "Item"
                }
            },
            itemIsInBasket() {
                return false
            },
        },
        localVue,
    })
    it('displays as expected', () => {
        expect(wrapper.html()).toMatchSnapshot()
    })
})
