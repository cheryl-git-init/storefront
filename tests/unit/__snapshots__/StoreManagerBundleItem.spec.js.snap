// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoreManagerBundleItem.vue displays as expected 1`] = `<span><span>Item</span></span>`;
